src/gui/qgsnewgeopackagelayerdialog.cpp

```diff
@@ -349,6 +349,13 @@ bool QgsNewGeoPackageLayerDialog::apply()
   OGRwkbGeometryType wkbType = static_cast<OGRwkbGeometryType>
                                ( mGeometryTypeBox->currentData( Qt::UserRole ).toInt() );
 
+  // z-coordinate & m-value.
+  if (mGeometryWithZCheckBox->isChecked())
```

Comment:
I'd prefer to see these split into 2 separate checkboxes. There's often times when you'd want Z or M without wanting both dimensions.